spec.html

```diff
@@ -38589,6 +38768,281 @@ <h1>JSON [ @@toStringTag ]</h1>
+          1. Return _weakRef_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-properties-of-the-weak-ref-constructor">
+      <h1>Properties of the WeakRef Constructor</h1>
+      <p>The WeakRef constructor:</p>
+      <ul>
+        <li>
+          has a [[Prototype]] internal slot whose value is the intrinsic object %Function.prototype%.
+        </li>
+        <li>has the following properties:</li>
+      </ul>
+
+      <emu-clause id="sec-weak-ref.prototype">
+        <h1>WeakRef.prototype</h1>
+        <p>
+          The initial value of `WeakRef.prototype` is the intrinsic %WeakRef.prototype% object.
+        </p>
+        <p>
+          This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *false* }.
+        </p>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-properties-of-the-weak-ref-prototype-object">
+      <h1>Properties of the WeakRef Prototype Object</h1>
+      <p>The WeakRef prototype object:</p>
+      <ul>
+        <li>is the intrinsic object <dfn>%WeakRef.prototype%</dfn>.</li>
+        <li>
+          has a [[Prototype]] internal slot whose value is %Object.prototype%.
+        </li>
+        <li>is an ordinary object.</li>
+        <li>does not have a [[WeakRefTarget]] internal slot.</li>
+      </ul>
+
+      <emu-clause id="sec-weak-ref.prototype.constructor" normative-optional>
+        <h1>WeakRef.prototype.constructor</h1>
+
+        <p>The initial value of `WeakRef.prototype.constructor` is the intrinsic object %WeakRef%.</p>
+
+        <p>This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</p>
+      </emu-clause>
+
+      <emu-clause id="sec-weak-ref.prototype.deref">
+        <h1>WeakRef.prototype.deref ( )</h1>
+        <p>The following steps are taken:</p>
+        <emu-alg>
+          1. Let _weakRef_ be the *this* value.
+          1. Perform ? RequireInternalSlot(_weakRef_, [[WeakRefTarget]]).
+          1. Let _target_ be the value of _weakRef_.[[WeakRefTarget]].
+          1. If _target_ is not ~empty~, then
+            1. Perform ! AddToKeptObjects(_target_).
+            1. Return _target_.
+          1. Return *undefined*.
+        </emu-alg>
+
+        <emu-note>
+          <p>
+          If the WeakRef returns a _target_ Object that is not *undefined*, then this _target_ object should not be garbage collected in that turn. The AddToKeptObjects operation makes sure read consistency is maintained.
+          </p>
+
+          <pre>
+            target = { foo: function() {} };
+            let weakRef = new WeakRef(target);
+
+            ... later ...
+
+            if (weakRef.deref()) {
+              weakRef.deref().foo();
+            }
+          </pre>
+
+          <p>
+          In the above example, if the first deref evaluates to true then the second deref cannot return *undefined*.
```

Comment:
"evaluates to true" is an odd thing to say, since the deref itself can only return an object ro `*undefined*`. Perhaps

```suggestion
          In the above example, if the first deref does not evaluate to *undefined* then the second deref cannot either.
```